### PR TITLE
io.cpp positive tests and remaining protocol.cpp tests

### PR DESCRIPTION
**Pull request by achaikou**

### CODE REVIEW

File: lib/src/io.cpp [hunk 1]
```diff
@@ -303,6 +309,7 @@ record& stream::at( int i, record& rec ) noexcept (false) {
 
             if (err) consistent = false;
             attributes.push_back( attrs );
+            types.push_back( type );
```
Comment (jokva) [inline lib/src/io.cpp]: What's going on here?

File: python/dlisio/ext/core.cpp [hunk 2]
```diff
@@ -418,10 +418,11 @@ PYBIND11_MODULE(core, m) {
...
             return b;
         })
-        .def( "extract", [](dl::stream& s, const std::vector< long long >& tells) {
+        .def( "extract", [](dl::stream& s,
+                            const std::vector< long long >& tells_indices) {
             std::vector< dl::record > recs;
-            recs.reserve( tells.size() );
-            for (auto i : tells) {
+            recs.reserve( tells_indices.size() );
+            for (auto i : tells_indices) {
```
Comment (jokva) [inline python/dlisio/ext/core.cpp]: I'm not sure this is actually a clarification. If it's indices anyways, why mix tells into it?

Comment (achaikou) [reply python/dlisio/ext/core.cpp]: I agree, it actually is just "indices" to extract, which would be more clear. But for some reason it has signature of "std::vector< long long >" (which corresponds to tells one) and I am unable to understand where we got it from. 
If that's not a problem, I would happily change it just to "indices".

File: lib/src/io.cpp (hunk 1 above)
Comment (achaikou) [reply lib/src/io.cpp]: Every record has a "type" field. There is a line further in the method that reads type from types vector (types.front()). But we never put current type there, so it was always reading "0" from empty vector before.

**Pull request merged**